save-deploy/README.md

```diff
@@ -183,19 +183,128 @@ Usually, not the whole stack is required for development. Application logic is p
    The resulting application settings may look like this: [screenshot](../info/img/github-oauth-app-settings.png).
 
 ## Local debugging
-You can run backend, orchestrator, preprocessor and frontend locally in IDE in debug mode.<br/>
-If you run on Windows, dependency `save-agent` is omitted because of problems with linking in cross-compilation.<br/>
-To run on Windows you need to compile `save-agent` on WSL and put `saveAgentDistroFilepath` to `%USERPROFILE%\.gradle\gradle.properties` <br/>
-For example:
+You can run backend, orchestrator, preprocessor and frontend locally in IDE in debug mode.
+
+#### Using `save-agent` executable on Windows
+
+If you run on Windows, dependency `save-agent` is omitted because of problems with linking in cross-compilation.
+To run on Windows, you need to build and package `save-agent` on WSL:
+
+```bash
+./gradlew :save-agent:copyAgentDistribution
+```
+
+and provide the path to the JAR archive which contains `save-agent.kexe` via the
+`saveAgentDistroFilepath` _Gradle_ property, by setting the above property
+either under project-specific `gradle.properties`, or, globally, under
+`%USERPROFILE%\.gradle\gradle.properties`, e.g.:
+
+```properties
+# gradle.properties
+saveAgentDistroFilepath=file:\\\\\\\\wsl$\\Ubuntu\\home\\username\\projects\\save-cloud\\save-agent\\build\\libs\\save-agent-0.3.0-alpha.0.48+1c1fd41-distribution.jar
+```
+
+Using forward slashes on Windows is allowed, too (_Gradle_ will understand such
+paths just fine):
+
+```properties
+# gradle.properties
+saveAgentDistroFilepath=file:////wsl$/Ubuntu/home/username/projects/save-cloud/save-agent/build/libs/save-agent-0.4.0-SNAPSHOT-distribution.jar
+```
+
+Once the _agent_ distribution is built and `saveAgentDistroFilepath` is set, you
+can run (on Windows):
+
+```bat
+gradlew.bat :save-backend:downloadSaveAgentDistro
+```
+
+Once the task completes, the _agent_ JAR can be found under
+`save-backend\build\agentDistro` directory.
 
-    saveAgentDistroFilepath=file:\\\\\\\\wsl$\\Ubuntu\\home\\username\\projects\\save-cloud\\save-agent\\build\\libs\\save-agent-0.3.0-alpha.0.48+1c1fd41-distribution.jar
+For the classpath changes to take effect:
+
+1. Reload the project from disk (_Project_ tool window in IDEA).
+1. Reload the project model (_Gradle_ tool window in IDEA).
+1. Re-start the _back-end_ application.
+
+Then verify that the agent is indeed available for download from the _back-end_.
+Incorrect output example (the binary is not found):
+
+```console
+$ curl -vvv -X POST http://localhost:5800/internal/files/download-save-agent --output save-agent.kexe
+*   Trying 127.0.0.1:5800...
```

Comment:
may be remove this information? I think people will see it when they will run it

Reply:
Addressed with cd49c63011f418cd1724a05e5a7be13c852c7ab8